### PR TITLE
[DreamHost] Added exclusion for links subdomain, added rule for blog...

### PR DESCRIPTION
**Pull request by galenide**
...subdomain, removed exclusion for transferapproval subdomain, expanded observed subdomains list

### CODE REVIEW

**Comment by galenide** [comment]
@jeremyn From what it appears, the dreamhost.com domain has a number of subdomains which do not support HTTPS (such as links and whoisweb) and for which the current DreamHost ruleset redirects to HTTPS. As such, this revised version of the ruleset may be of some importance as far as reducing the chance of users encountering errors due to HTTPS Everywhere.

**Comment by J0WI** [comment]
This seems easy to convert from a wildcard rule to a list of targets.
@galenide would you like to do this? This simplifies the rule a lot.

**Comment by J0WI** [comment]
See also #7313

File: src/chrome/content/rules/DreamHost.xml [hunk 1]
```diff
@@ -28,39 +28,38 @@
...
 
 
-	<securecookie host="^(?:.*\.)?dreamhost\.com$" name=".+" />
+	<test url="http://media-dh-com.objects.dreamhost.com/mp4/player.swf" />
+
+
+	<securecookie host="^(\.?(abuse|atmail|discussion|gifts|media|panel|roundcube|signup|transferapproval|webftp|webmail|www))\.dreamhost\.com$"
+			name=".+" />
+	<securecookie host="(^|\.)objects\.dreamhost.com$"
+			name=".+" />
```
Comment (J0WI) [inline src/chrome/content/rules/DreamHost.xml]: You can also use `<securecookie host=".+"	name=".+" />` to match all targets.

File: src/chrome/content/rules/DreamHost.xml [hunk 2]
```diff
@@ -28,39 +28,38 @@
...
+	<target host="www.dreamhost.com" />
+	<target host="abuse.dreamhost.com" />
+	<target host="atmail.dreamhost.com" />
+	<target host="blog.dreamhost.com" />
+	<target host="discussion.dreamhost.com" />
+	<target host="gifts.dreamhost.com" />
+	<target host="help.dreamhost.com" />
+	<target host="media.dreamhost.com" />
+	<target host="objects.dreamhost.com" />
+	<target host="*.objects.dreamhost.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/DreamHost.xml]: Wildcard Hosts need at least 3 test urls.

**Comment by J0WI** [comment]
Thanks!

**Pull request merged**